### PR TITLE
more doxygen work

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #77 
Part of #70 

More doxygen work...

Only documentation changes in this PR - no code changes.

### CODE REVIEW

**Comment by edwardhartnett** [comment]
I'll hit the doxygen again tomorrow morning and finish this pass through.

After that I'll figure out what should be in every user_guide.md document that is currently empty.

**Review by Hang-Lei-NOAA — APPROVED** [review]

File: src/grbindex/grbindex.f [hunk 1]
```diff
@@ -1,81 +1,81 @@
...
+!> PROGRAM HISTORY LOG:
+!> -  92-11-22  IREDELL
+!> -  94-06-08  EBISUZAKI - ELIMINATE ISHELL CALLS
+!> -  94-08-26  IREDELL   - 40 BYTE PDS EXTENSION
+!> -  95-10-31  IREDELL   - CONSIDERABLY REDUCE I/O
+!> -  96-10-31  IREDELL   - AUGMENTED OPTIONAL DEFINITIONS TO BYTE 320
+!> - 1999-04-27  GILBERT   - Changed CALL EXIT(N) to CALL ERREXIT(N) so that
+!>                         the proper error return value is passed back to
+!>                         the shell.
+!> - 2012-07-31  VUONG     - CHANGED HOSTNAME TO HOSTNAM
```
Comment (BrianCurtis-NOAA) [inline src/grbindex/grbindex.f]: This isn't consistent with our other PHL's.

File: src/grbindex/grbindex.f [hunk 2]
```diff
@@ -197,51 +197,51 @@ SUBROUTINE WRGI1H(LUGI,NLEN,NNUM,CGB)
...
+!
+! SUBPROGRAM: NCBASE         LOCATE BASENAME OF A FILE
+!   PRGMMR: IREDELL          ORG: W/NMC23     DATE: 93-11-22
+!
+! ABSTRACT: THIS SUBPROGRAM LOCATES THE CHARACTER NUMBER AFTER THE LAST 
+!   IN A CHARACTER STRING.  FOR UNIX FILENAMES, THE CHARACTER NUMBER
+!   RETURNED MARKS THE BEGINNING OF THE BASENAME OF THE FILE.
+!
+! PROGRAM HISTORY LOG:
+!   93-11-22  IREDELL
```
Comment (BrianCurtis-NOAA) [inline src/grbindex/grbindex.f]: We've typically removed this if there's only the initial author.

File: src/grbindex/grbindex.f (hunk 1 above)
Comment (edwardhartnett) [reply src/grbindex/grbindex.f]: Fixed.

File: src/grbindex/grbindex.f (hunk 2 above)
Comment (edwardhartnett) [reply src/grbindex/grbindex.f]: Fixed.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Pull request merged**